File: app/javascript/src/graphql/mutations/DeleteTasks.ts
import {createMutationHook, gql} from '../graphqlHelpers';
import {deleteTasks, deleteTasksVariables} from './__generated__/deleteTasks';

const deleteTasks = createMutationHook<deleteTasks, deleteTasksVariables>(gql`
  mutation deleteTasks($taskIds: [ID!]!) {
    deleteTasks(taskIds: $taskIds) {
      success
      taskIds
    }
  }
`);

export default deleteTasks;

File: app/graphql/mutations/delete_tasks.rb
class Mutations::DeleteTasks < GraphQL::Schema::Mutation
  argument :taskIds, [ID], required: true

  field :errors, [Types::FormErrorType], null: false
  field :success, Boolean, null: false
  field :task_ids, [ID], null: false

  def resolve(task_ids: nil)
    response = { errors: [] }

    task_ids.each { |id| Task.find(id).destroy! }

    response[:success] = true
    # returning task ids to update the DOM
    response[:task_ids] = task_ids
    response
  end
end
